src/main/java/org/eclipse/yasson/model/PropertyModel.java

```diff
@@ -136,18 +133,40 @@ private PropertyCustomization introspectCustomization(Property property, JsonbCo
         final AnnotationIntrospector introspector = jsonbContext.getAnnotationIntrospector();
         final PropertyCustomizationBuilder builder = new PropertyCustomizationBuilder();
         //drop all other annotations for transient properties
-        if (introspector.isTransient(property)) {
-            builder.setJsonbTransient(true);
-            return builder.buildPropertyCustomization();
+        EnumSet<AnnotationTarget> transientInfo = introspector.getJsonbTransientCategorized(property);
+        if (transientInfo.size() != 0) {
+            if(transientInfo.contains(AnnotationTarget.GETTER)) {
```

Comment:
can it be just transientInfo.setReadTransient(transientInfo.contains(AnnotationTarget.GETTER)) ?